File: analyze_excel.py
import openpyxl

file_path = r"c:\Users\i0215099\Desktop\ANTI_TEST\정삭 장비 우선 순위(HSP HM2J AH2J 10T 15T).xlsx"
print(f"Loading {file_path}")

try:
    wb = openpyxl.load_workbook(file_path, data_only=True)
    ws = wb.active
    
    print("Row 1 (Header):")
    for cell in ws[1][:10]:
        fill = cell.fill.start_color.rgb if cell.fill and cell.fill.start_color else "None"
        font = cell.font.color.rgb if cell.font and cell.font.color else "None"
        bold = cell.font.bold if cell.font else "None"
        h_align = cell.alignment.horizontal if cell.alignment else "None"
        v_align = cell.alignment.vertical if cell.alignment else "None"
        borders = []
        if cell.border:
            if cell.border.top.style: borders.append("top")
            if cell.border.bottom.style: borders.append("bottom")
            if cell.border.left.style: borders.append("left")
            if cell.border.right.style: borders.append("right")
        print(f"  {cell.coordinate} ({cell.value}): fill={fill}, font_color={font}, bold={bold}, align={h_align}/{v_align}, borders={'-'.join(borders)}")

    print("\nRow 2 (Data):")
    for cell in ws[2][:10]:
        fill = cell.fill.start_color.rgb if cell.fill and cell.fill.start_color else "None"
        font = cell.font.color.rgb if cell.font and cell.font.color else "None"
        bold = cell.font.bold if cell.font else "None"
        h_align = cell.alignment.horizontal if cell.alignment else "None"
        v_align = cell.alignment.vertical if cell.alignment else "None"
        borders = []
        if cell.border:
            if cell.border.top.style: borders.append("top")
            if cell.border.bottom.style: borders.append("bottom")
            if cell.border.left.style: borders.append("left")
            if cell.border.right.style: borders.append("right")
        print(f"  {cell.coordinate} ({cell.value}): fill={fill}, font_color={font}, bold={bold}, align={h_align}/{v_align}, borders={'-'.join(borders)}")

    print("\nRow 3 (Data):")
    for cell in ws[3][:10]:
        fill = cell.fill.start_color.rgb if cell.fill and cell.fill.start_color else "None"
        font = cell.font.color.rgb if cell.font and cell.font.color else "None"
        bold = cell.font.bold if cell.font else "None"
        h_align = cell.alignment.horizontal if cell.alignment else "None"
        v_align = cell.alignment.vertical if cell.alignment else "None"
        borders = []
        if cell.border:
            if cell.border.top.style: borders.append("top")
            if cell.border.bottom.style: borders.append("bottom")
            if cell.border.left.style: borders.append("left")
            if cell.border.right.style: borders.append("right")
        print(f"  {cell.coordinate} ({cell.value}): fill={fill}, font_color={font}, bold={bold}, align={h_align}/{v_align}, borders={'-'.join(borders)}")

except Exception as e:
    print(f"Error: {e}")

File: analyze_excel.js
const ExcelJS = require('exceljs');

async function main() {
    const workbook = new ExcelJS.Workbook();
    await workbook.xlsx.readFile('정삭 장비 우선 순위(HSP HM2J AH2J 10T 15T).xlsx');
    const worksheet = workbook.worksheets[0];

    for (let i = 1; i <= 3; i++) {
        console.log(`\nRow ${i}:`);
        const row = worksheet.getRow(i);
        row.eachCell({ includeEmpty: true }, (cell, colNumber) => {
            if (colNumber > 10) return;
            const fgColor = cell.fill && cell.fill.fgColor ? cell.fill.fgColor.argb : "None";
            const fontColor = cell.font && cell.font.color ? cell.font.color.argb : "None";
            const bold = cell.font ? !!cell.font.bold : false;
            const fontSize = cell.font ? cell.font.size : "None";
            const hAlign = cell.alignment ? cell.alignment.horizontal : "None";
            const vAlign = cell.alignment ? cell.alignment.vertical : "None";

            let borders = [];
            if (cell.border) {
                if (cell.border.top) borders.push("top");
                if (cell.border.bottom) borders.push("bottom");
                if (cell.border.left) borders.push("left");
                if (cell.border.right) borders.push("right");
            }
            console.log(`  ${cell.address} (${cell.value}): fill=${fgColor}, font_color=${fontColor}, bold=${bold}, size=${fontSize}, align=${hAlign}/${vAlign}, borders=${borders.join('-')}`);
        });
    }
}
main().catch(console.error);
